### PR TITLE
feat: start and stop node

### PR DESCRIPTION
**Pull request by panjan**

### CODE REVIEW

File: lib/node_starter/killer.rb [hunk 1]
```diff
@@ -0,0 +1,58 @@
...
+  # class killing running node process
+  class Killer
+    attr_reader :build_id
+
+    def initialize(build_id)
+      @build_id = build_id
+    end
+
+    def shutdown
+      node = Node.find_by build_id: @build_id
```
Comment (lksv) [inline lib/node_starter/killer.rb]: did you mean `find_by!`  ?

File: lib/node_starter/killer.rb [hunk 2]
```diff
@@ -0,0 +1,58 @@
...
+      node.status = :aborting
+      node.save!
```
Comment (lksv) [inline lib/node_starter/killer.rb]: `node.update_colmun(:status, :aborting)`

File: lib/node_starter/killer.rb [hunk 3]
```diff
@@ -0,0 +1,58 @@
...
+
+      sleep 120 if shutdown_using_api node.uri # TODO: retry logic with increasing interval
```
Comment (lksv) [inline lib/node_starter/killer.rb]: its quit long to wait. Is any way to check that is already done?

File: lib/node_starter/killer.rb [hunk 4]
```diff
@@ -0,0 +1,58 @@
...
+      kill_process node.pid if running? node.pid
+      force_kill_process node.pid if running? node.pid
+    end
+
+    private
+
+    def kill_process(pid)
+      NodeStarter.logger.debug("Checking running node to be aborted build_id=#{@build_id}")
+      5.times.with_index do |i|
```
Comment (lksv) [inline lib/node_starter/killer.rb]: `5.times do |i|`
the `5` should come form config.

**Comment by lksv** [comment]
I start with code review, but I have to stop now due to laack of time. It is a lot of code probably I would suggest a lot of architectural changes. Please check:
* what is better to do through model method and what is service object
* why is a each request started in new-new thread (I guess Bunny start one thread and you the second in it).
* the TODO is usually handled in pull request than in the code.

**Comment by petrSchreiber** [comment]
Code is fine from my side, cannot merge

**Pull request merged**